### PR TITLE
rtl88x2bu: 2022-12-17 -> 2023-02-24

### PR DESCRIPTION
**Pull request by otavio**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by Ralith — APPROVED** [review]

File: pkgs/os-specific/linux/rtl88x2bu/default.nix [hunk 1]
```diff
@@ -25,6 +25,9 @@ stdenv.mkDerivation rec {
 
   preInstall = ''
     mkdir -p "$out/lib/modules/${kernel.modDirVersion}/kernel/net/wireless/"
+
+    # Install 88x2bu.conf so it works in Linux 6.2 or newer.
+    install -D -m 64 $src/88x2bu.conf $out/etc/modprobe.d/88x2bu.conf
```
Comment (NickCao) [inline pkgs/os-specific/linux/rtl88x2bu/default.nix:30]: Does this actually work?

Comment (otavio) [reply pkgs/os-specific/linux/rtl88x2bu/default.nix]: Fixed.

File: pkgs/os-specific/linux/rtl88x2bu/default.nix [hunk 2]
```diff
@@ -25,6 +25,9 @@ stdenv.mkDerivation rec {
 
   preInstall = ''
     mkdir -p "$out/lib/modules/${kernel.modDirVersion}/kernel/net/wireless/"
+
+    # Install 88x2bu.conf so it works in Linux 6.2 or newer.
+    install -D -m 644 $src/88x2bu.conf $out/etc/modprobe.d/88x2bu.conf
```
Comment (NickCao) [inline pkgs/os-specific/linux/rtl88x2bu/default.nix:30]: I meant that is this file automatically added into `/etc/modprobe.d` (without adding it to `boot.extraModprobeConfig`.

Comment (otavio) [reply pkgs/os-specific/linux/rtl88x2bu/default.nix:30]: I want to thank you for raising this. But, indeed, it doesn't work. I tried to find a way to blacklist it, but it doesn't seem to exist, except by using a NixOS module.

I've removed this install command, and the user must block the module by hand.

**Review by NickCao — APPROVED** [review]

**Pull request merged**